pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -95,6 +95,14 @@ func UpdatedConditionUnknownErrorMessageWithClock(clock clock.Clock, condition g
 	return UpdatedConditionWithClock(clock, condition, gardencorev1beta1.ConditionUnknown, gardencorev1beta1.ConditionCheckError, message, codes...)
 }
 
+// BuildConditions builds and returns the conditions using the given conditions as a base,
+// by first removing all conditions with the given types and then merging the given conditions (which must be of the same types).
+func BuildConditions(baseConditions []gardencorev1beta1.Condition, newConditions []gardencorev1beta1.Condition, conditionTypes []gardencorev1beta1.ConditionType) []gardencorev1beta1.Condition {
```

Comment:
Would it not make sense to rename `conditionTypes` to something else?

`baseConditions` and `newConditions` give the reader a hint as to what will happen with them. From `conditionTypes` it is rather hard to understand that the conditions of these types will be removed.